### PR TITLE
Enable recommended `eslint-plugin-jsdoc` rules

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

This upgrades and enables recommended `eslint-plugin-jsdoc` rules.

### CODE REVIEW

File: packages/realm/realm.node
Comment (takameyer) [inline packages/realm/realm.node:1]: This binary should probably not be committed

File: packages/realm/.eslintrc.json [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
     "@typescript-eslint/explicit-function-return-type": "off",
     "no-console": "error",
-    "sort-imports": ["warn", { "ignoreDeclarationSort": true }]
+    "sort-imports": ["warn", { "ignoreDeclarationSort": true }],
+    "jsdoc/check-tag-names": "off",
+    "jsdoc/require-jsdoc": "off",
+    "jsdoc/require-returns": "off",
+    "jsdoc/require-yields": "off",
+    "jsdoc/require-param": "off",
+    "jsdoc/no-undefined-types":
```
Comment (takameyer) [inline packages/realm/.eslintrc.json]: Need a value here. (linter job is failing)

**Review by takameyer — APPROVED** [review]
The `.eslint.json` is broken in `packages/realm`.  Other than that, LGTM

**Comment by kraenhansen** [comment]
I've filed an issue with the JSDocs plugin for some of the warnings introduced by merging this PR: https://github.com/gajus/eslint-plugin-jsdoc/issues/1119

**Pull request merged**